### PR TITLE
Fixes #34610 - Entitlement report double counts consumed subs

### PR DESCRIPTION
**Pull request by ianballou**
Updates the Katello Subscription Entitlement report to use a new, corrected method for calculating how many entitlements a host consumes.

### CODE REVIEW

**Comment by ianballou** [comment]
Intended to be tested with https://github.com/Katello/katello/pull/10013

**Review by ezr-ondrej — DISMISSED** [review]
This is ok to merge once the katello PR gets in :+1:

File: app/views/unattended/report_templates/subscription_-_entitlement_report.erb [hunk 1]
```diff
@@ -39,8 +39,8 @@ require:
           'Products': host_products_names(host),
           'Subscription Name': sub_name(pool),
           'Subscription Type': pool.type,
-          'Subscription Quantity': pool.quantity,
-          'Subscriptions Consumed': pool.consumed,
+          'Subscription Total Quantity': pool.quantity,
+          'Subscriptions Consumed': host.filtered_entitlement_quantity_consumed(pool),
```
Comment (jeremylenz) [inline app/views/unattended/report_templates/subscription_-_entitlement_report.erb]: Should we rename this column to make it clear that it's specific to the host? Users who already use this report may get confused when the number changes..

Comment (ianballou) [reply app/views/unattended/report_templates/subscription_-_entitlement_report.erb]: I updated the names.

**Comment by ianballou** [comment]
[test katello]

**Comment by ianballou** [comment]
The Katello PR is merged!

**Review by jeremylenz — APPROVED** [review]
LGTM :+1: 

thanks @ianballou !

**Pull request merged**